src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PensionAnnexDateStampService.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.Loader;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.interactive.form.PDAcroForm;
+import org.apache.pdfbox.pdmodel.interactive.form.PDField;
+import org.apache.pdfbox.pdmodel.interactive.form.PDTextField;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.finrem.caseorchestration.error.DocumentStorageException;
+import uk.gov.hmcts.reform.finrem.caseorchestration.error.StampDocumentException;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.FinremMultipartFile;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseDocument;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.document.Document;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.evidence.FileUploadResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.evidencemanagement.EvidenceManagementDownloadService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.evidencemanagement.EvidenceManagementUploadService;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static org.springframework.http.MediaType.APPLICATION_PDF_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.DocumentManagementService.CONVERTER;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class PensionAnnexDateStampService {
+    private final EvidenceManagementUploadService emUploadService;
+    private final EvidenceManagementDownloadService emDownloadService;
+    private final GenericDocumentService genericDocumentService;
+    static final String FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME = "Date the court made/varied/discharged an order";
+    static final String DATE_STAMP_PATTERN = "dd MMMM yyyy";
+    static final String DEFAULT_PDTYPE_FONT_HELV = "Helv";
+    static final String DEFAULT_PDTYPE_PREFIX = "/";
+    static final String DEFAULT_PDTYPE_POSTFIX = " 10 Tf 0 g";
+
+    public CaseDocument appendApprovedDateToDocument(CaseDocument document,
+                                                     String authToken,
+                                                     LocalDate approvalDate,
+                                                     String caseId) throws Exception {
+        log.info("Adding date stamp to Pension Sharing Annex : {}", document);
```

Comment:
Consider if CaseDocument isn't log safe.  For instance, if document_filename contains anything that could identify an individual.

Reply:
If you decide on something log safe, consider including it on your DocumentStorageException below.  Consider including the passed in case id too.